Lib/functools.py

```diff
@@ -1000,6 +1026,8 @@ def __get__(self, instance, owner=None):
                 val = cache.get(self.attrname, _NOT_FOUND)
                 if val is _NOT_FOUND:
                     val = self.func(instance)
+                    if inspect.iscoroutinefunction(self.func):
```

Comment:
`inspect.isawaitable(val)` check covers more cases, please use it.

Reply:
This would change the behaviour of

```python
class MyAwaitable:
    def __await__(self):
        print("called")

class Foo:
    @cached_property
    def get_awaitable(self):
        return MyAwaitable()
```

Currently this would cache the `MyAwaitable` object and calls `__await__`. If we use `isawaitable` here, `MyAwaitable` would be wrapped and only awaited once.

IMO `iscoroutinefunction` is more correct here.